### ISSUE TITLE
Jupyterlab extension not detecting nodejs on Ubuntu

### ISSUE THREAD

**Issue by stuaxo** [kiteco/issue-tracker#414]
**Required Information**
To help us triage your issue effectively, please make sure you've completed these steps before opening this issue. Issues with not enough information may be deprioritized or ignored.

- [x] [Read our guide](https://help.kite.com/article/96-contact-us-with-logs) on how to report an issue.
- [x] Include your Kite application logs, _taken while you are experiencing your issue_.
https://s3.console.aws.amazon.com/s3/object/kite-client-logs/prod/126208/kite_local/client.log.2020-10-03_04-02-16-AM.bak.gz

- [x] If reporting a performance problem, include your resource usage logs, _taken while you are experiencing your issue_.
- [x] If reporting an issue with Kite's editor plugin, include which editor you are using and the version, and any accompanying logs from your editor's console.

**Describe the bug**
Installing the jupyter extension via pip is not working for me under Ubuntu 20.04.
It looks like the extension doesn't check for nodejs installed using the `deb` from ubuntus package manager.

Shell transcript:

```bash
$ nodejs --version
v10.19.0
$ jupyter labextension install "@kiteco/jupyterlab-kite"
An error occured.
ValueError: Please install nodejs >=10.0.0 before continuing. nodejs may be installed using conda or directly from the nodejs website.
See the log file for details:  /tmp/jupyterlab-debug-acp47tr4.log

$ which nodejs
/usr/bin/nodejs

$ dpkg -s nodejs
Package: nodejs
Status: install ok installed
Priority: optional
Section: javascript
Installed-Size: 154
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Multi-Arch: foreign
Version: 10.19.0~dfsg-3ubuntu1
Replaces: nodejs-legacy
Depends: libc6 (>= 2.4), libnode64 (= 10.19.0~dfsg-3ubuntu1)
Recommends: ca-certificates, nodejs-doc
Suggests: npm
Conflicts: nodejs-legacy
Description: evented I/O for V8 javascript - runtime executable
 Node.js is a platform built on Chrome's JavaScript runtime for easily
 building fast, scalable network applications. Node.js uses an
 event-driven, non-blocking I/O model that makes it lightweight and
 efficient, perfect for data-intensive real-time applications that run
 across distributed devices.
 .
 Node.js is bundled with several useful libraries to handle server
 tasks:
 .
 System, Events, Standard I/O, Modules, Timers, Child Processes, POSIX,
 HTTP, Multipart Parsing, TCP, DNS, Assert, Path, URL, Query Strings.
Homepage: http://nodejs.org/
Original-Maintainer: Debian Javascript Maintainers <pkg-javascript-devel@lists.alioth.debian.org>
```

**Screenshots**

**Comment by its-dhung** [comment]
Did you use `conda` to install JupyterLab? If so, you unfortunately may need to install nodejs with `conda` for this to work.

Since this isn't an issue related to Kite, I'm going to close it.

**Comment by stuaxo** [comment]
I am not a conda user.

I'm not sure why it requires conda?


Whenever I try to use conda, it is too hard to use with the libraries that I tend to use, so I give up, I'm pretty happy with debian, and pip, or pip on  Windows for last 10 years.


If anything I'd like to open a feature request for this to work with the default linux package manager.

**Comment by its-dhung** [comment]
You don't need `conda` to use Kite actually. I was just wondering since the error message said "nodejs may be installed using conda or directly from the nodejs website".

Do you have 2 installations of nodejs on your system? Wondering if JupyterLab is detecting the wrong one to use.

**Comment by its-dhung** [comment]
After doing some digging, I wonder if [this issue](https://github.com/jupyterlab/jupyterlab/issues/3640) in the official JupyterLab repo is what you're seeing.

It seems like you can either

1. Add the desired nodejs to your `PATH` variable or
2. Run `sudo apt install -y nodejs npm`

**Comment by stuaxo** [comment]
Hi,
   Thanks for all the help, that sent me down the right, eg... path.
I had these installed and on the PATH, but it turned out I had an older version installed via nvm.

I unistalled nvm and removed it from the PATH, opened a new shell and then was able to install the lab extension.

Seems to be a theme for second or third languages for programmers, I've fixed a lot of bugs where people have many versions of python installed for instance.

Cheers
Stuart
